tests/src/reloadFunctionalTest/kotlin/org/jetbrains/compose/reload/tests/VerificationTests.kt

```diff
@@ -84,11 +84,60 @@ class VerificationTests {
         }
     }
 
+    @HotReloadTest
+    @QuickTest
+    fun `illegal code change - change local variables in main`(fixture: HotReloadTestFixture) = fixture.runTest {
+        val code = fixture.initialSourceCode(
+            """
+                import androidx.compose.material.Text
+                import org.jetbrains.compose.reload.test.*
+                
+                fun main() {
+                    val str1 = "Hello"
+                    screenshotTestApplication {
+                        Text(str1 + "world!")
+                    }
+                }
+            """.trimIndent()
+        )
+
+        fixture.checkScreenshot("0-initial")
+
+        fixture.runTransaction {
+            /*
+            Legal Change: replace the code inside the composable function
+            */
+            code.replaceText(""""world!"""", """" world!"""")
```

Comment:
When changing code, make sure to use more 'different-looking' texts: 
The image diff algorithm is OK with anti-aliasing working different and therefore is lenient towards 'some difference'. It would be better for the human eye and the algorithm if we use something like 'Foo' & 'Bar' or 'A' 'B' or